### PR TITLE
feat: added definitions for es-get-iterator, iterate-iterator, and iterate-value

### PR DESCRIPTION
**Pull request by JasonHK**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/es-get-iterator/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Type definitions for es-get-iterator 1.1
+// Project: https://github.com/ljharb/es-get-iterator#readme
+// Definitions by: Jason Kwok <https://github.com/JasonHK>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.6
+
+interface Iterable<T, TReturn = any, TNext = undefined> {
```
Comment (ljharb) [inline types/es-get-iterator/index.d.ts:7]:
```suggestion
interface Iterable<T, TReturn = unknown, TNext = undefined> {
```

File: types/iterate-iterator/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Type definitions for iterate-iterator 1.0
+// Project: https://github.com/ljharb/iterate-iterator#readme
+// Definitions by: Jason Kwok <https://github.com/JasonHK>
```
Comment (ljharb) [inline types/iterate-iterator/index.d.ts:3]:
```suggestion
// Definitions by: Jason Kwok <https://github.com/JasonHK>
// Definitions by: Jordan Harband <ljharb@gmail.com>
```

File: types/iterate-value/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// Type definitions for iterate-value 1.0
+// Project: https://github.com/ljharb/iterate-value#readme
+// Definitions by: Jason Kwok <https://github.com/JasonHK>
```
Comment (ljharb) [inline types/iterate-value/index.d.ts:3]:
```suggestion
// Definitions by: Jason Kwok <https://github.com/JasonHK>
// Definitions by: Jordan Harband <ljharb@gmail.com>
```

Comment (JasonHK) [reply types/iterate-value/index.d.ts:3]: Done.

File: types/iterate-iterator/index.d.ts (hunk 2 above)
Comment (JasonHK) [reply types/iterate-iterator/index.d.ts:3]: Done.

File: types/es-get-iterator/index.d.ts (hunk 1 above)
Comment (JasonHK) [reply types/es-get-iterator/index.d.ts]: Done.

File: types/es-get-iterator/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// Type definitions for es-get-iterator 1.1
+// Project: https://github.com/ljharb/es-get-iterator#readme
+// Definitions by: Jason Kwok <https://github.com/JasonHK>
```
[the remark this answers is not in the document]
Comment (JasonHK) [reply types/es-get-iterator/index.d.ts:3]: Done.

**Review by ljharb — APPROVED** [review]

**Review by orta — APPROVED** [review]

**Comment by JasonHK** [comment]
Ready to merge

**Pull request merged**